fcli-core/fcli-aviator-common/src/main/java/com/fortify/cli/aviator/grpc/AviatorGrpcClientHelper.java

```diff
@@ -1,23 +1,67 @@
 package com.fortify.cli.aviator.grpc;
 
-import com.fortify.cli.aviator.config.AviatorLoggerImpl;
-import com.fortify.cli.common.progress.helper.IProgressWriter;
+import com.fortify.cli.aviator.config.IAviatorLogger;
+import io.grpc.CompressorRegistry;
+import io.grpc.DecompressorRegistry;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.concurrent.TimeUnit;
+
 public class AviatorGrpcClientHelper {
     private static final Logger LOG = LoggerFactory.getLogger(AviatorGrpcClientHelper.class);
-    private static final int DEFAULT_PORT = 9090;
+    private static final int DEFAULT_TIMEOUT_SECONDS = 30;
 
-    public static AviatorGrpcClient createClient(String url) {
-        try{
-            String[] parts = url.split(":");
-            String host = parts[0];
-            int port = parts.length > 1 ? Integer.parseInt(parts[1]) : DEFAULT_PORT;
-            LOG.debug("Creating gRPC client for host: {}, port: {}", host, port);
-            return new AviatorGrpcClient(host, port, 10, null);
-        } catch (NumberFormatException e) {
-            throw new IllegalArgumentException("Invalid port in URL: " + url, e);
+    public static AviatorGrpcClient createClient(String url, IAviatorLogger logger) {
+        if (url == null || url.trim().isEmpty()) {
+            throw new IllegalArgumentException("URL cannot be null or empty");
```

Comment:
Better to throw an Aviator/Fcli SimpleException, as that gives us the possibility to output a more friendly error message rather than dumping a full stack trace.